perfact/zodbsync/commands/upload.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+
+import os
+import sys
+
+from ..subcommand import SubCommand
+from ..zodbsync import mod_format
+
+
+PY2 = (sys.version_info.major == 2)
+
+META_TEMPLATES = {
+    'folder': {
+        'props': [],
+        'type': 'Folder',
+        'title': '',
+    },
+    'js': {
+        'props': [
+            [
+                ('id', 'content_type'),
+                ('type', 'string'),
+                ('value', 'application/javascript')
+            ]
+        ],
+        'type': 'File',
+        'title': '',
+    },
+    'css': {
+        'props': [
+            [
+                ('id', 'content_type'),
+                ('type', 'string'),
+                ('value', 'text/css')
+            ]
+        ],
+        'type': 'File',
+        'title': '',
+    }
+}
```

Comment:
I still think this might be written more compactly. Just a gut feeling, not a strong objection. But let me try it:

    def create_template(type, content_type=None)
        result = {'type': type, 'title': ''}
        if content_type is not None:
            result['props'] = [[('id', 'content_type'), ('type', 'string'), ('value', content_type)]]
        return result

    META_TEMPLATES = {
        'folder': create_template('Folder'),
        'js': create_template('File', 'application/javascript'),
        'css': create_template('File', 'text/css'),
    }

What do you think?

Reply:
Good one, will copy & paste :smile: